mesonbuild/interpreterbase/decorators.py

```diff
@@ -702,7 +719,7 @@ def log_usage_warning(self, tv: str, location: T.Optional['mparser.BaseNode']) -
         ]
         if self.extra_message:
             args.append(self.extra_message)
-        mlog.warning(*args, location=location)
+        mlog.warning(*args, location=location, fatal=not always)
```

Comment:
This should just be dropped. The default is True, but we're setting it to False *iff* it is an `always_warn`.

This means that a deprecation for something that is always deprecated and always wrong and always should be replaced... does NOT trigger --fatal-meson-warnings, even though it really should.